lib/TableGen/DialectType.cpp

```diff
@@ -44,6 +44,28 @@ bool DialectType::init(raw_ostream &errs, GenDialectsContext &context,
   m_summary = record->getValueAsString("summary");
   m_description = record->getValueAsString("description");
 
+  if (auto *dag =
+          cast<DagInit>(record->getValue("representation")->getValue())) {
+    if (cast<DefInit>(dag->getOperator())->getDef()->getName() ==
+        "repr_struct") {
+      m_structBacked = true;
+
+      if (dag->getNumArgs() != 1) {
+        errs << "'repr_struct' expects exactly one type argument\n";
+        return false;
+      }
+      m_structSentinelBitWidth =
+          llvm::cast<llvm::IntInit>(
+              llvm::cast<llvm::DagInit>(dag->getArg(0))->getArg(0))
+              ->getValue();
+    }
+  }
+  if (auto *p = record->getValue("structPrefix"))
+    m_structPrefix = record->getValueAsString("structPrefix").str();
+  else
+    m_structPrefix =
+        (m_dialectRec->getValueAsString("name").str() + "." + m_mnemonic + ".");
```

Comment:
getValue will never return nullptr here.

More generally -- the mnemonic is what we use for the TargetExtType. For consistency, it's better to just remove struct prefix altogether and use the mnemonic as the prefix.